h/services/group_list.py

```diff
@@ -68,21 +68,31 @@ def associated_groups(self, user=None):
         """
         Return a list of groups associated with a user.
 
-        Relevant groups include groups the user is a
-        creator/moderator/member of.
+        Include groups that this group is a member of as well as public groups
+        that this user is the creator of.
 
         If the user is None it returns an empty list.
         """
         if user is None:
             return []
-        world_readable_groups = [
-            group
-            for group in self._readable_by_world_groups(user, None)
-            if group.creator == user or user in group.members
-        ]
-        private_groups = self.private_groups(user)
```

Comment:
From the PR:

> This PR rewrites the logic for associated_groups (used by h.panels.navbar) such that it makes a single query to retrieve groups instead of many. 

How many queries did you think this was making before? By my count this was making one for `_readable_by_world_groups`, one for each world readable group at the authority (!) to load `group.members` and then another one to load the `user.groups` relation.

Reply:
Yep. A lot. New Relic is suggesting that on _average_ there are about 100 user selects per page load. That's just the user part. Yeah.